ideas/apq.h

```diff
@@ -0,0 +1,285 @@
+using io_service = boost::asio::io_service;
+
+
+/**
+* Should we use defined type of pg_connect_handle, or some customized type and
+* template for connection?
+*   PGconn                       - pointer to native connection handle
+*   std::function<void(PGconn*)> - finalization function, e.g. destructor for single 
+*                                  connection or recycling operation for pool.
+*/
+using pg_connect_handle = std::shared_ptr<PGconn>; //, std::function<void(PGconn*)>>;
+
+
+/**
+* We can try to use standard error_code, but we will loose the error message
+* from database. But if we will use the custom error_code we are forced to
+* implement our own yield_context and async_result infrastructure. This perspective
+* is not attractive. We can provide some kind of yield_context support, but with
+* message loss in case of error_code bound to yield (i.e. yield[ec])
+*/
+using error_code = boost::system::error_code; // std:: / custom ?
+
+/**
+* Connection - represents binding of hative handle an io_service (io_context) object
+* Should we allow to copy connection itself, or the object must be move-only?
+*/
+struct connection {
+	io_service& ios;
```

Comment:
No tabs in this project, please.

Reply:
I'll fix it. Didn't cate about tabs for the prototype. :)